server-spi-private/src/main/java/org/keycloak/models/utils/ModelToRepresentation.java

```diff
@@ -265,10 +266,21 @@ public static UserRepresentation toBriefRepresentation(UserModel user) {
         rep.setEnabled(user.isEnabled());
         rep.setEmailVerified(user.isEmailVerified());
         rep.setFederationLink(user.getFederationLink());
+        addAttributeToBriefRep(user, rep, TEMP_ADMIN_ATTR_NAME);
 
         return rep;
     }
 
+    public static void addAttributeToBriefRep(UserModel user, UserRepresentation userRep, String attributeName) {
+        String userAttributeValue = user.getFirstAttribute(attributeName);
+        if (Boolean.parseBoolean(userAttributeValue)) {
+            if (userRep.getAttributes() == null) {
+                userRep.setAttributes(new HashMap<>());
+            }
+            userRep.getAttributes().put(attributeName, Collections.singletonList(user.getFirstAttribute(attributeName)));
+        }
+    }
+
```

Comment:
IMO, if we want to expose the method to the outside, it should be more generic - not to accept only booleans.

Reply:
@mabartos good catch, this is just a utility method, there is no need to expose it outside the ModelToRepresentation class, so the `private` access modifier should be used here.

To your main point, I guess we can do the following:
```
    private static <T> void addAttributeToBriefRep(UserModel user, UserRepresentation userRep, String attributeName, Function<String, T> parser) {
        String userAttributeValue = user.getFirstAttribute(attributeName);
        T parsedValue = parser.apply(userAttributeValue);
        if (parsedValue != null) {
            if (userRep.getAttributes() == null) {
                userRep.setAttributes(new HashMap<>());
            }
            userRep.getAttributes().put(attributeName, Collections.singletonList(userAttributeValue));
        }
    }
```
Then the method can be called like this:
`addAttributeToBriefRep(user, rep, TEMP_ADMIN_ATTR_NAME, Boolean::parseBoolean);`

Wdyt?

Reply:
@mabartos Good catch!

I don't think we need to parse/cast the types at all. All attrs are just strings, no? We just put them from model to rep, nothing more. Or am I missing something?

Reply:
@vmuzikar you're correct,  I guess just simple check `userAttributeValue != null` should cover all the possible cases.

Let me quickly try it.

Reply:
@mabartos @vmuzikar I [applied](https://github.com/keycloak/keycloak/compare/cc0d74765ae20a74ad66ee4877da5b037f9abbf8..69d74ce5315cefac4338530f988647c051a34175) all of Martin's suggestions.